python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -371,6 +371,45 @@ def test_chain_dag(ray_start_regular, num_actors):
     compiled_dag.teardown()
 
 
+def test_execution_timeout(ray_start_regular):
```

Comment:
Can you add tests that we don't time out if the result is ready, even though the timeout is "expired".

Reply:
Added `test_get_with_zero_timeout`, and similarly `test_execution_with_tiny_timeout`. We need a tiny (instad of 0) timeout because `header_sem` may be acquired by the reader, and it is not always a success for the first writer attempt.

Reply:
Hmm I think the `test_get_with_zero_timeout` is what I was looking for here. Probably we don't need `test_execution_with_tiny_timeout` (that looks likely to flake to me).

Reply:
Removed `test_execution_with_tiny_timeout`.